### ISSUE TITLE
Paging Through Shop Pages returns the Wrong # of Products

### ISSUE THREAD

**Issue by martinsmagic** [woocommerce/woocommerce#13398]
## EXPLANATION OF THE ISSUE

This is happening in 2.7.0-RC1 - I have not seen it before.   The # of products being returned in the shop pages is wrong.  Instead of being 10 per page it appears almost random for each page.

## STEPS TO REPRODUCE THE ISSUE

I have created a stripped down version of my site and you can see it here:  https://beta.qualitymagicsales.com/

It is easy to reproduce just click on the links below and you'll see how each shop page has a "random" number of products instead of 10.

The first page shows 8 products (should be 10) 
https://beta.qualitymagicsales.com/

The second page shows 3:
https://beta.qualitymagicsales.com/page/2/

The third page shows 3:
https://beta.qualitymagicsales.com/page/3/

I have removed pretty much every plugin etc.  The live site is here for comparison:

http://www.qualitymagicsales.com/

Thanks.

andy

## SYSTEM STATUS REPORT

```
Grab the system status report from WooCommerce > System Status and paste it here.
`
### WordPress Environment ###

Home URL: https://beta.qualitymagicsales.com
Site URL: https://beta.qualitymagicsales.com
WC Version: 2.7.0
Log Directory Writable: ✔
WP Version: 4.7.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.4.7 (Ubuntu)
PHP Version: 5.5.9-1ubuntu4.21
PHP Post Max Size: 100 MB
PHP Time Limit: 180
PHP Max Input Vars: 1620
cURL Version: 7.35.0
OpenSSL/1.0.1f

SUHOSIN Installed: –
MySQL Version: 5.5.54
Max Upload Size: 100 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.7.0
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (5) ###

Rename wp-login.php: by Ella Iseulde Van Dorpe – 2.5.5
WooCommerce - Store Toolkit: by Visser Labs – 1.6.9
WooCommerce: by Automattic – 2.7.0-RC1
WP Security Audit Log: by WP White Security – 2.6
Clef: by Clef – 2.6.1

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
simple (simple)
variable (variable)


### WC Pages ###

Shop base: #31094 - /
Cart: #827 - /cart/
Checkout: #828 - /checkout/
My account: #830 - /my-account/

### Theme ###

Name: Divi
Version: 3.0.34
Author URL: http://www.elegantthemes.com
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –

### Templates ###

thumbnail: 
medium: 
medium_large: 
large: 
shop_thumbnail: 
shop_catalog: 
shop_single: 
et-pb-post-main-image: 
et-pb-post-main-image-fullwidth: 
et-pb-portfolio-image: 
et-pb-portfolio-module-image: 
et-pb-portfolio-image-single: 
et-pb-gallery-module-image-portrait: 
`

<!--
BEFORE POSTING YOUR ISSUE
- These comments won't show up when you submit the issue.
- Try to add as much detail as possible. Be specific!
- GitHub issues ARE NOT FOR SUPPORT! If you have questions, use the either:
  - https://wordpress.org/support/plugin/woocommerce for general support
  - https://woocommerce.com/my-account/tickets/ for premium plugin support
- If you're requesting a new feature, explain why you'd like it to be added.
- Search this repository (top of the page) for the issue and it has not been fixed or reported already.

Before logging BUGS, ensure you:
- Use the latest stable release of WooCommerce.
- Disabled all plugins to ensure it's a core bug and not a plugin issue.
- Switched to Twenty Twelve theme to ensure it's a core bug and not a theme issue.
-->

**Comment by mikejolley** [comment]
Did you run the data upgrader in admin?

**Comment by martinsmagic** [comment]
Hi Mike - yes I did.

Is there a setting I can check in the DB to ensure it ran properly?


Thanks,

andy

Andy Martin, Ph.D. www.martinsmagic.com [http://www.martinsmagic.com]
Magician & Collector www.qualitymagicsales.com [http://www.qualitymagicsales.com]
andy@martinsmagic.com www.qualitymagicsites.com [http://www.qualitymagicsites.com]
On 3/2/2017 8:21:27 AM, Mike Jolley <notifications@github.com> wrote:
Did you run the data upgrader in admin?

**Comment by mikejolley** [comment]
2 things then; try a default WP theme, and try 'saving' one of your missing products to see if it appears.

**Comment by martinsmagic** [comment]
Mike - I see what the problem is.

It is not showing the out of stock items.  But instead of filling in the blanks with other in-stock products it just omits them.

If I uncheck the: "Hide out of stock items from the catalog" then they all come back.

So hopefully that helps.

This is using Twenty Twelve theme.

Thanks,

andy

Andy Martin, Ph.D. www.martinsmagic.com [http://www.martinsmagic.com]
Magician & Collector www.qualitymagicsales.com [http://www.qualitymagicsales.com]
andy@martinsmagic.com www.qualitymagicsites.com [http://www.qualitymagicsites.com]
On 3/2/2017 8:47:43 AM, Mike Jolley <notifications@github.com> wrote:
2 things then; try a default WP theme, and try 'saving' one of your missing products to see if it appears.

**Comment by mikejolley** [comment]
Sounds like the data upgrade script did not update the out of stock products with the correct taxonomy. I'll double check the updater.

**Comment by martinsmagic** [comment]
ah yes - that makes perfect sense.  Thanks Mike.

andy

Andy Martin, Ph.D. www.martinsmagic.com [http://www.martinsmagic.com]
Magician & Collector www.qualitymagicsales.com [http://www.qualitymagicsales.com]
andy@martinsmagic.com www.qualitymagicsites.com [http://www.qualitymagicsites.com]
On 3/2/2017 9:09:45 AM, Mike Jolley <notifications@github.com> wrote:
Sounds like the data upgrade script did not update the out of stock products with the correct taxonomy. I'll double check the updater.

**Comment by martinsmagic** [comment]
beautiful - thanks for fixing it so quickly.

andy

Andy Martin, Ph.D. www.martinsmagic.com [http://www.martinsmagic.com]
Magician & Collector www.qualitymagicsales.com [http://www.qualitymagicsales.com]
andy@martinsmagic.com www.qualitymagicsites.com [http://www.qualitymagicsites.com]
On 3/2/2017 9:59:45 AM, Mike Jolley <notifications@github.com> wrote:
Closed #13398 [https://github.com/woocommerce/woocommerce/issues/13398] via 5637f04 [https://github.com/woocommerce/woocommerce/commit/5637f043e52dfb5db3b1a05a2bd9716257582413].

**Comment by martinsmagic** [comment]
Hi Mike,

just downloaded the latest RC version and this problem has not been fixed.

https://beta.qualitymagicsales.com/

The out of stock items are still showing up as phantom items.

It clearly is something to do with the updater and out of stock items, but your fix does not appeared to have work.

Thanks,

andy
